app/src/main/kotlin/com/ayatk/biblio/ui/home/ranking/TopRankingFragment.kt

```diff
@@ -16,34 +16,96 @@
 
 package com.ayatk.biblio.ui.home.ranking
 
+import android.arch.lifecycle.ViewModelProviders
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import com.ayatk.biblio.databinding.FragmentRankingBinding
+import com.ayatk.biblio.di.ViewModelFactory
+import com.ayatk.biblio.util.Result
+import com.ayatk.biblio.util.ext.observe
 import dagger.android.support.DaggerFragment
+import timber.log.Timber
 import javax.inject.Inject
 
 class TopRankingFragment : DaggerFragment() {
 
-  lateinit var binding: FragmentRankingBinding
-
   @Inject
-  lateinit var viewModel: TopRankingViewModel
+  lateinit var viewModelFactory: ViewModelFactory
+
+  private val viewModel: TopRankingViewModel by lazy {
+    ViewModelProviders.of(this, viewModelFactory).get(TopRankingViewModel::class.java)
+  }
+
+  lateinit var binding: FragmentRankingBinding
 
   override fun onCreateView(
       inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
   ): View? {
     binding = FragmentRankingBinding.inflate(inflater, container, false)
-    binding.viewModel = viewModel
-    viewModel.start()
+    binding.setLifecycleOwner(this)
+
+    bind()
 
     return binding.root
   }
 
-  override fun onDetach() {
-    viewModel.destroy()
-    super.onDetach()
+  private fun bind() {
```

Comment:
**[detekt]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
Prefer splitting up complex methods into smaller, easier to understand methods.

Reply:
🙆